src/backend/mi2/mi2.ts

```diff
@@ -478,18 +478,18 @@ export class MI2 extends EventEmitter implements IBackend {
 									}
 								} else
 									this.log("log", JSON.stringify(parsed));
-							} else if (record.type == "notify") {
-								if (record.asyncClass == "thread-created") {
+							} else if (record.type === "notify") {
+								if (record.asyncClass === "thread-created") {
 									this.emit("thread-created", parsed);
-								} else if (record.asyncClass == "thread-exited") {
+								} else if (record.asyncClass === "thread-exited") {
 									this.emit("thread-exited", parsed);
 								}
 							}
 						}
 					});
 					handled = true;
 				}
-				if (parsed.token == undefined && parsed.resultRecords == undefined && parsed.outOfBandRecord.length == 0)
+				if (parsed.token === undefined && parsed.resultRecords === undefined && parsed.outOfBandRecord.length === 0)
```

Comment:
need to also check the types more here to see if this could not happen otherwise, so better to leave this out for now